File: Programmers/Level2/삼각달팽이/삼각달팽이.py
def solution(n):
    matrix = [[0] * n for _ in range(n)]
    x, y = -1, 0
    num = 1
    
    for i in range(n):
        for _ in range(i, n):
            if i % 3 == 0:
                x += 1
            elif i % 3 == 1:
                y += 1
            else:
                x -= 1
                y -= 1
            
            matrix[x][y] = num
            num += 1
            
    return [value for inner_matrix in matrix for value in inner_matrix if value != 0]

File: Programmers/Level2/삼각달팽이/삼각달팽이.js
function solution(n) {
    const matrix = [...Array(n)].map(() => [...Array(n).fill(0)]);
    let [x, y] = [-1, 0];
    let num = 1;

    for (let i = 0; i < n; i++) {
        for (let _ = i; _ < n; _++) {
            if (i % 3 == 0) x += 1;
            else if (i % 3 == 1) y += 1;
            else if (i % 3 == 2) {
                x -= 1;
                y -= 1;
            }
            matrix[x][y] = num;
            num += 1;
        }
    }

    return matrix.map((e) => e.filter((v) => v !== 0)).flat();
}
